packages/@sanity/form-builder/src/inputs/InvalidValueInput/UntypedValueInput.tsx

```diff
@@ -38,56 +48,84 @@ function UnsetItemButton({
-  const itemValue = {...value, _type: validTypes[0]}
-  return (
-    <DefaultButton onClick={() => onChange(PatchEvent.from(unset()), itemValue)} color="danger">
-      Remove value
-    </DefaultButton>
-  )
+  const itemValue = useMemo(() => ({...value, _type: validTypes[0]}), [validTypes, value])
+
+  const handleClick = useCallback(() => onChange(PatchEvent.from(unset()), itemValue), [
+    itemValue,
+    onChange,
+  ])
+
+  return <Button onClick={handleClick} tone="critical" text="Unset value" />
 }
 
 /**
  * When the value does not have an `_type` property,
  * but the schema has a named type
  */
-export function UntypedValueInput({validTypes, value, onChange}: Props) {
+export function UntypedValueInput({validTypes, value, onChange}: UntypedValueInputProps) {
   const isSingleValidType = validTypes.length === 1
   const isHoistedType = schema.has(validTypes[0])
-  const fix = isSingleValidType ? (
-    <SetMissingTypeButton onChange={onChange} targetType={validTypes[0]} value={value} />
-  ) : null
-
-  const message = (
-    <>
-      Encountered an object value without a <code>_type</code> property.
-      {isSingleValidType && !isHoistedType && (
-        <div>
-          Either remove the <code>name</code> property of the object declaration, or set{' '}
-          <code>_type</code> property on items.
-        </div>
-      )}
-      {!isSingleValidType && (
-        <div>
-          The following types are valid here according to schema:{' '}
-          <ul>
-            {validTypes.map((validType) => (
-              <li key={validType}>
-                <code>{validType}</code>
-              </li>
-            ))}
-          </ul>
-        </div>
-      )}
-      <h4>object</h4>
-      <pre className={styles.inspectValue}>{JSON.stringify(value, null, 2)}</pre>
-      {fix}
-      {fix && ' '}
-      <UnsetItemButton onChange={onChange} validTypes={validTypes} value={value} />
-    </>
-  )
 
-  return <Warning heading="Content is missing _type" message={message} />
+  return (
+    <Alert
+      status="warning"
+      title={
+        <>
+          Property value missing <code>_type</code>
+        </>
+      }
+    >
+      <Details open={__DEV__} title={<>Developer info</>}>
+        <Stack space={3}>
+          <Text as="p" muted size={1}>
+            Encountered an object value without a <code>_type</code> property.
+          </Text>
+
+          {isSingleValidType && !isHoistedType && (
+            <Text as="p" muted size={1}>
+              Either remove the <code>name</code> property of the object declaration, or set{' '}
+              <code>_type</code> property on items.
+            </Text>
+          )}
+
+          {!isSingleValidType && (
+            <Text as="p" muted size={1}>
+              The following types are valid here according to schema:
+            </Text>
+          )}
+
+          {!isSingleValidType && (
+            <ul>
```

Comment:
Same as above (use `Stack`)?

Reply:
Fixed in https://github.com/sanity-io/sanity/commit/3f01b74e1c002950ab13b3e1bfcefc32d51a9417